### ISSUE TITLE
False negative for undefined-variable when attempting to access class scope from list comprehension with double for loop

### ISSUE THREAD

**Issue by Jamie-** [PyCQA/pylint#5861]
### Bug description

```py
# pylint: disable=missing-docstring, disallowed-name, too-few-public-methods

class Foo:

    bar = 6.0
    foo = {"a": 1, "b": 2}
    baz = [
        a*b
        for a, b in foo.items()
        for i in range(bar)
    ]
```

Without the double for loop in the comprehension, pylint performs as expected.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:10:23: E0602: Undefined variable 'bar' (undefined-variable)
```

### Expected behavior

No warning

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.7.3 (default, Jan 22 2021, 20:04:44) 
[GCC 8.3.0]

(also failing in Pylint 2.7.0, haven't tested further back)
```

### OS / Environment

Debian 10.11 (Buster) and Ubuntu 20.04 LTS

**Comment by mbyrnepr2** [comment]
Hi thanks for this @Jamie- . I've taken a look and it appears you are running into a Python issue covered here: https://bugs.python.org/issue3692.

The same behaviour reported in this issue is seen if you run your code with Python instead & not Pylint.

**Comment by DanielNoord** [comment]
@mbyrnepr2 I guess this mean we can close this right? This does indeed not execute and the warning therefore seems valid.

**Comment by mbyrnepr2** [comment]
I believe so @DanielNoord. Just FYI I don't have the permissions to close it.

**Comment by Jamie-** [comment]
The example given was a simplified example from a much larger codebase which is why I missed the float issue. Assume that's changed to an int, this code indeed does not execute under Python 3 but does execute (and as expected) under Python 2.

Sadly a regression in Python I guess, we'd not noticed this yet as we're currently porting this huge codebase from Python 2 to 3 and this is and edge case we need to fixup. Thanks for the Python bug link.
